roles/cdi/tasks/provision.yml

```diff
@@ -15,48 +15,29 @@
   when: ns.rc != 0
         and platform == "openshift"
 
+- name: Render CDI operator
+  template:
+    src: "cdi-operator.yaml.j2"
+    dest: "/tmp/cdi-operator.yaml"
+
+- name: Create CDI Operator
+  command: "{{ cluster_command }} apply -f /tmp/cdi-operator.yaml --validate=false"
+
 - name: Render {{ cdi_namespace }} ResourceQuota deployment yaml
   template:
-    src:  cdi-resourcequota.yml
-    dest: /tmp/cdi-provision-resourcequota.yml
+    src: "cdi-resourcequota.yml.j2"
+    dest: "/tmp/cdi-provision-resourcequota.yml"
 
 - name: Create {{ cdi_namespace }} ResourceQuota
   command: "{{ cluster_command }} apply -f /tmp/cdi-provision-resourcequota.yml -n {{ cdi_namespace }}"
 
-- name: Check for cdi-controller.yaml.j2 template in {{ cdi_template_dir }}
-  stat:
-    path: "{{ cdi_template_dir }}/cdi-controller.yaml.j2"
-  register: byo_template
-
-- name: Render CDI deployment yaml from template in {{ cdi_template_dir }}
+- name: Render CDI operator resources
```

Comment:
Need to wait for the cdi-operator to be 'Running' before we create the CR.  Otherwise, it will be a race since the CRD may not have been created yet.

Reply:
```diff
+ - name: Wait until cdi-operator is running
+  shell: "{{ cluster_command }} -n {{ cdi_namespace }} get deployment | grep cdi-operator | awk \'{ if ($2 == $5) print \"0\"; else print \"1\"}\'"
+  register: result
+  until: result.stdout == "0"
+  retries: 24
+  delay: 10
```

Reply:
or could use wait again.

Reply:
done, thanks